Add support for assuming roles from web identities.

markdouthwaite opened the pull request:
Please ensure the following before submitting a PR:

 - [x] if suggesting code changes or improvements, [open an issue](https://github.com/cloudyr/aws.signature/issues/new) first
 - [x] for all but trivial changes (e.g., typo fixes), add your name to [DESCRIPTION](https://github.com/cloudyr/aws.signature/blob/master/DESCRIPTION)
 - [x] for all but trivial changes (e.g., typo fixes), documentation your change in [NEWS.md](https://github.com/cloudyr/aws.signature/blob/master/NEWS.md) with a parenthetical reference to the issue number being addressed
 - [x] if changing documentation, edit files in `/R` not `/man` and run `devtools::document()` to update documentation
 - [x] add code or new test files to [`/tests`](https://github.com/cloudyr/aws.signature/tree/master/tests/testthat) for any new functionality or bug fix
 - [x] make sure `R CMD check` runs without error before submitting the PR

This PR adds support for [assuming roles from web identities](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html) as requested in [#62](https://github.com/cloudyr/aws.signature/issues/62)

Commit:
Add assume_role_with_web_identity

Commit:
Update assume_role.R

Commit:
Update DESCRIPTION

Commit:
Update DESCRIPTION

Commit:
Build package

Commit:
Build package - again!

Commit:
Update assume_role.R

Commit:
Update assume_role.R

Commit:
Update assume_role.R

Commit:
Update locate_credentials.R

Commit:
Update locate_credentials.R

Commit:
Update locate_credentials.R

Commit:
Add Dockerfile & explicit namespace.

Commit:
Update assume_role.R

Commit:
Update assume_role.R

Commit:
Update file reader line

Commit:
Add print statements

Commit:
unpack AWS STS response

markdouthwaite committed:
Update locate_credentials.R

markdouthwaite committed:
Update assume_role.R

markdouthwaite committed:
Delete Dockerfile

markdouthwaite committed:
Add timeout and updated session_name

markdouthwaite committed:
Remove timeout

markdouthwaite committed:
Update assume_role.R

markdouthwaite committed:
Update assume_role.R

markdouthwaite committed:
Update assume_role.R

markdouthwaite committed:
Update assume_role.R

markdouthwaite committed:
Update locate_credentials.R

markdouthwaite committed:
Merge pull request #1 from PeakBI/feature/support-assume-role-with-web-identity

Feature/support assume role with web identity

markdouthwaite committed:
Update DESCRIPTION

markdouthwaite committed:
Update NEWS.md

jon-mago reviewed (commented):
Hi there

Thanks for taking the time to write this Mark, and for your patience while I got back to you!

I have some comments on the changes which I've put on the relevant lines, though I'm happy to talk about any and all of them.

One other comment on this, and I appreciate this is probably a limitation of how the package is currently written, but I think this code is going to result in the package creating new temporary credentials for each request to AWS.  This doesn't seem 100% ideal.  But I don't have an easy fix for that.

lukedyer-uipath committed:
add verbosity flag to assume_role

lukedyer-peak commented on R/assume_role.R:
```diff
@@ -0,0 +1,45 @@
+  content <- httr::content(response)
+  
+  if (httr::status_code(response) == 200) {
+    message("Successfully fetched token.")
```
👍  addressed this in [1eac62d](https://github.com/cloudyr/aws.signature/pull/63/commits/1eac62d025921f4812c07ef213f740b5830d7564)

Commit:
replace httr with curl in assume_role_with_web_identity

Commit:
update DESCRIPTION

markdouthwaite committed:
Merge pull request #2 from PeakBI/remove-httr

Remove `httr` dependency

lukedyer-uipath committed:
fix typo in authors list

lukedyer-uipath committed:
fix typo in Imports

lukedyer-uipath committed:
update man files

lukedyer-uipath committed:
change environment variable credential priority

This favors default environment variables over the web identity provider
environment variables.
This was done to address package author's PR comment.

lukedyer-uipath committed:
refactor variable names to be more descriptive

lukedyer-uipath committed:
clarify code comments

lukedyer-uipath committed:
add contributor to authors

lukedyer-uipath committed:
strip whitespace

lukedyer-uipath committed:
bug: fix variable typo

lukedyer-uipath committed:
make logging message more explicit

lukedyer-uipath committed:
fix linting error

lukedyer-uipath committed:
Merge pull request #3 from PeakBI/credential-ordering

Change credential ordering

lukedyer-peak commented on DESCRIPTION:
```diff
@@ -1,21 +1,24 @@
     digest,
-    base64enc
+    base64enc,
+    httr
```
Address this to just use `curl` and `jsonlite`

lukedyer-peak commented:
@jon-mago - we've addressed all the review comments now 😄
